### PR TITLE
Skip merging identical slices instead of erroring

### PR DESCRIPTION
**Pull request by wenovus**
Currently, merging identical slices when calling `MergeStruct*` causes an error, complaining that the slices aren't unique per RFC6020.

This behaviour is inconsistent with what we do for unions, scalars, pointers, and binary, where we skip merging, and simply keep the identical value.

This PR implements this more consistent behaviour, which is the (1, 1) behaviour combination described in #492.

### CODE REVIEW

**Comment by wenovus** [comment]
@robshakir note I didn't add a `MergeOpt` here, because I currently think skipping the merge makes more sense as the default behaviour than erroring out, although I do see the argument that people might assume a behaviour that's not what they intended, where having the `MergeOpt` would make sense.

As the PR currently stands, I think we have a good support for the base case of RFC6020. Special de-duping behaviour, or RFC7950 support can be added with `MergeOpt` in the future.

File: ygot/struct_validation_map.go [hunk 1]
```diff
@@ -820,6 +820,10 @@ func copySliceField(dstField, srcField reflect.Value, opts ...MergeOpt) error {
 	}
 
 	if _, ok := srcField.Interface().([]Annotation); !ok {
+		if diff := cmp.Diff(srcField.Interface(), dstField.Interface()); diff == "" {
```
Comment (robshakir) [inline ygot/struct_validation_map.go:823]: Does `cmp.Diff` do something different to `cmp.Equal` here? I can't remember why we used `cmp.Diff` originally.

**Review by robshakir — APPROVED** [review]
LGTM, thanks for this change -- I think you're right that things brings things back to what we'd expect for equal fields. (I think the original reason that we had an error here was that we expected that values *would not* be populated in both cases, and rather we'd error if the field was populated in both -- but slices weren't covered when the fix to allow for equal values was allowed).

**Comment by robshakir** [comment]
Actually, one other thing -- can you update the docstring [here](https://godoc.corp.google.com/pkg/google3/third_party/golang/ygot/ygot/ygot#MergeStructs) since I think we need to say that if the slices are equal, then they are not merged.

File: ygot/struct_validation_map.go (hunk 1 above)
Comment (wenovus) [reply ygot/struct_validation_map.go:823]: https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff
```
It returns an empty string if and only if Equal returns true for the same input values and options.
```

Comment (robshakir) [reply ygot/struct_validation_map.go:825]: Given that you don't use the `diff` here, I suggest just changing this to be `if !cmp.Equal()`.

**Pull request merged**